aiida/engine/processes/builder.py

```diff
@@ -238,6 +244,43 @@ def __init__(self, process_class: Type['Process']):
         self._process_spec = self._process_class.spec()
         super().__init__(self._process_spec.inputs)
 
+    def _repr_pretty_(self, p, _):  # pylint: disable=invalid-name
+        """Pretty representation for in the IPython console and notebooks."""
+
+        def format_inputs(inputs, level=0):
```

Comment:
I would just extract this as a separate staticmethod on the class:

```python
@staticmethod
def _format_inputs(inputs: dict, level: int=0) -> str:
```

Reply:
Had to make it a `classmethod` since it's recursive.